crates/modelardb_server/src/context.rs

```diff
@@ -851,6 +840,20 @@ mod tests {
         assert_eq!(schema, test::model_table_metadata().schema)
     }
 
+    async fn parse_and_create_table(context: &Context, sql: &str) -> Result<()> {
+        match parser::tokenize_and_parse_sql_statement(sql)? {
+            ModelarDbStatement::CreateNormalTable { name, schema } => {
+                context.create_normal_table(name, schema, sql).await
+            }
+            ModelarDbStatement::CreateModelTable(model_table_metadata) => {
+                context.create_model_table(model_table_metadata, sql).await
+            }
+            _ => Err(ModelarDbServerError::InvalidArgument(
```

Comment:
I guess it will no longer be a problem once we no longer use `is_err()` and check for the error message instead. My concern is that in a test where we have `is_err()`, for example when testing that we cannot create an existing normal table, the function might return the wrong error but the test will still succeed. 

`parse_and_create_table()` should still return a result but returning an error when the statement is not `CreateNormalTable` or `CreateModelTable` seems problematic.

Reply:
Good point, I have changed the error if the statement is not one of the expected types to `unreachable!()` as it should never be reached.